pkg/aws/aws_client/client.go

```diff
@@ -106,3 +115,32 @@ func (client *AWSClient) CloudFormation() *cloudformation.Client {
 func (client *AWSClient) ELB() *elb.Client {
 	return client.ElbClient
 }
+
+func GrantValidAccessKeys(userName string) (*AccessKeyMod, error) {
+	var cre aws.Credentials
+	var keysMod *AccessKeyMod
+	var err error
+	retryTimes := 3
+	for retryTimes > 0 {
+		if cre.AccessKeyID != "" {
+			break
+		}
+		client, err := CreateAWSClient(userName, CON.DefaultAWSRegion)
+		if err != nil {
+			return nil, err
+		}
+
+		cre, err = client.AWSConfig.Credentials.Retrieve(client.ClientContext)
+		if err != nil {
+			return nil, err
+		}
+		log.LogInfo(">>> Access key grant successfully")
+
+		keysMod = &AccessKeyMod{
```

Comment:
@yingzhanredhat I think you should `return` here rather than going back into the retry loop. So for example:

```
cre, err = client.AWSConfig.Credentials.Retrieve(client.ClientContext)
if err != nil {
	return nil, err
} else {
 log.LogInfo(">>> Access key grant successfully")
 return &AccessKeyMod{
  AccessKeyId:     cre.AccessKeyID,
  SecretAccessKey: cre.SecretAccessKey,
 },nil
}
```

Reply:
Ah, I missed that you exit the retry loop via:

```
if cre.AccessKeyID != "" {
			break
		}
```